testsuite/features/support/env.rb

```diff
@@ -130,26 +135,80 @@ def capybara_register_driver
 # Define the current feature scope
 Before do |scenario|
   $feature_scope = scenario.location.file.split(%r{(\.feature|/)})[-2]
+  if $timeout_failure_count >= $timeout_threshold
+    warn "Timeout failure threshold reached, stopping test suite."
+    exit(1)
```

Comment:
What about raising an error instead?

Reply:
Cucumber.wants_to_quit = true seems to do nothing.